### PR TITLE
Migrated web-sockets.md to null safety

### PR DESCRIPTION
**Pull request by miquelbeltran**
cc. @sfshaza2 @RedBrogdon

### CODE REVIEW

File: src/docs/cookbook/networking/web-sockets.md [hunk 1]
```diff
@@ -36,9 +38,9 @@ from the server and push messages to the server.
 In Flutter, use the following line to
 create a `WebSocketChannel` that connects to a server:
 
-<!-- skip -->
+<?code-excerpt "lib/main.dart (connect)" replace="/_channel/channel/g"?>
 ```dart
-final channel = IOWebSocketChannel.connect('ws://echo.websocket.org');
+final channel = IOWebSocketChannel.connect('wss://echo.websocket.org');
```
Comment (miquelbeltran) [inline src/docs/cookbook/networking/web-sockets.md:43]: Not sure about the difference here, but the complete example was using `wss`

File: src/docs/cookbook/networking/web-sockets.md [hunk 2]
```diff
@@ -53,14 +55,14 @@ In this example, use a [`StreamBuilder`][]
 widget to listen for new messages, and a
 [`Text`][] widget to display them.
 
-<!-- skip -->
+<?code-excerpt "lib/main.dart (StreamBuilder)" replace="/_channel/channel/g"?>
 ```dart
 StreamBuilder(
-  stream: widget.channel.stream,
```
Comment (miquelbeltran) [inline src/docs/cookbook/networking/web-sockets.md:59]: by moving the `channel` variable inside the sample widget state, I removed the need to have the `widget` property here

File: src/docs/cookbook/networking/web-sockets.md [hunk 3]
```diff
@@ -123,25 +125,26 @@ class MyApp extends StatelessWidget {
...
         title: title,
-        channel: IOWebSocketChannel.connect('wss://echo.websocket.org'),
       ),
     );
   }
 }
 
 class MyHomePage extends StatefulWidget {
   final String title;
-  final WebSocketChannel channel;
```
Comment (miquelbeltran) [inline src/docs/cookbook/networking/web-sockets.md:134]: I moved the channel directly inside the _MyHomePageState, simplifying a bit the example

File: src/docs/cookbook/networking/web-sockets.md [hunk 4]
```diff
@@ -160,13 +163,11 @@ class _MyHomePageState extends State<MyHomePage> {
                 decoration: InputDecoration(labelText: 'Send a message'),
               ),
             ),
+            SizedBox(height: 24),
```
Comment (miquelbeltran) [inline src/docs/cookbook/networking/web-sockets.md:166]: I have removed the Padding and instead introduced a SizedBox here, so the example is a bit cleaner

File: src/docs/cookbook/networking/web-sockets.md [hunk 5]
```diff
@@ -83,7 +85,7 @@ To send data to the server,
 `add()` messages to the `sink` provided
 by the `WebSocketChannel`.
 
-<!-- skip -->
+<?code-excerpt "lib/main.dart (add)" replace="/_channel/channel/g;/_controller.text/'Hello!'/g"?>
```
Comment (miquelbeltran) [inline src/docs/cookbook/networking/web-sockets.md:88]: it's maybe a bit of overkill all the replacements here but.. ¯\\\_(ツ)_/¯

File: src/docs/cookbook/networking/web-sockets.md (hunk 1 above)
Comment (domesticmouse) [reply src/docs/cookbook/networking/web-sockets.md:43]: `wss` is the tls encrypted version of `ws`.

**Review by domesticmouse — APPROVED** [review]

**Pull request merged**